pipelines/build/common/config_regeneration.groovy

```diff
@@ -332,13 +357,28 @@ class Regeneration implements Serializable {
         Map<String, ?> params = config.toMap().clone() as Map
         params.put("JOB_NAME", jobName)
         params.put("JOB_FOLDER", jobFolder)
+        params.put("SCRIPT_PATH", scriptPath)
 
         params.put("GIT_URI", gitUri)
         params.put("GIT_BRANCH", gitBranch)
 
+        params.put("DEFAULTS_JSON", JsonOutput.prettyPrint(JsonOutput.toJson(DEFAULTS_JSON)))
         params.put("BUILD_CONFIG", config.toJson())
 
-        def create = context.jobDsl targets: "pipelines/build/common/create_job_from_template.groovy", ignoreExisting: false, additionalParameters: params
+        // If we are not using default lib or script param values, be sure to update the initial downstream job script file
+        if (libraryPath != DEFAULTS_JSON['importLibraryScript']) {
+            params.put("CUSTOM_LIBRARY_LOCATION", libraryPath)
+        }
+        if (baseFilePath != DEFAULTS_JSON["baseFileDirectories"]["downstream"]) {
+            params.put("CUSTOM_BASEFILE_LOCATION", baseFilePath)
+        }
+
+        // Pass in checkout creds if needs be
+        if (checkoutCreds != "") {
```

Comment:
can checkout creds be null?

Reply:
Not really as the code will check for the existence of it later and conditional statements are not exactly reliable inside the template files